webview-ui/src/components/account/AccountView.tsx

```diff
@@ -165,30 +183,38 @@ export const ClineAccountView = () => {
 		async (event: any) => {
 			const newValue = (event.target as VSCodeDropdownChangeEvent["target"]).value || "personal"
 			const organizationId = newValue === "personal" ? undefined : newValue
+			const requestKey = organizationId || "personal"
 
 			if (newValue === dropdownValue) {
 				return // No change, do nothing
 			}
 
-			try {
-				console.info("Changing selection to:", newValue)
+			// Request deduplication check
+			if (isOrganizationSwitchingRef.current === requestKey) {
+				return
+			}
 
-				// Send the change to the server
-				AccountServiceClient.setUserOrganization({ organizationId })
+			// Mark this organization switch as in progress
+			isOrganizationSwitchingRef.current = requestKey
+			console.log(`[ORG_SWITCH_DEDUP] Request allowed - setting lock for "${requestKey}"`)
+
+			try {
+				await AccountServiceClient.setUserOrganization({ organizationId })
```

Comment:
```suggestion
				AccountServiceClient.setUserOrganization({ organizationId })
```
Do we need the await if we are not depending on the response? This should help with the speed too